Yi-Huang Han - Wed Oct 24 21:55:47 PDT 2001
 * Fixed MOD music volume when looping

diff -ru libmikmod-3.1.12.orig/playercode/mplayer.c libmikmod-3.1.12/playercode/mplayer.c
--- libmikmod-3.1.12.orig/playercode/mplayer.c	2009-10-05 00:19:59.000000000 -0700
+++ libmikmod-3.1.12/playercode/mplayer.c	2009-10-05 00:44:35.000000000 -0700
@@ -3019,7 +3019,7 @@
 {
 	MUTEX_LOCK(vars);
 	if (pf)
-		pf->volume=(volume<0)?0:(volume>128)?128:volume;
+		pf->volume=pf->initvolume=(volume<0)?0:(volume>128)?128:volume;
 	MUTEX_UNLOCK(vars);
 }
 
